src/librustc_mir/tcx/pattern.rs

```diff
@@ -153,38 +151,38 @@ impl<'tcx> Mirror<'tcx> for PatNode<'tcx> {
                 let hi = const_eval::eval_const_expr(cx.tcx, hi);
                 let hi = Literal::Value { value: hi };
                 PatternKind::Range { lo: lo, hi: hi }
-            },
+            }
 
             hir::PatEnum(..) | hir::PatIdent(..) | hir::PatQPath(..)
-                if pat_is_resolved_const(&cx.tcx.def_map, self.pat) =>
-            {
+                if pat_is_resolved_const(&cx.tcx.def_map, self.pat) => {
                 let def = cx.tcx.def_map.borrow().get(&self.pat.id).unwrap().full_def();
                 match def {
                     def::DefConst(def_id) | def::DefAssociatedConst(def_id) =>
                         match const_eval::lookup_const_by_id(cx.tcx, def_id, Some(self.pat.id)) {
-                            Some(const_expr) => {
-                                let opt_value =
+                        Some(const_expr) => {
+                            let opt_value =
```

Comment:
That's exactly what's happening. We try to place it on the same line and simply add a line break when it does not fit. We should reformat the expression.